client.go

```diff
@@ -1892,6 +1947,61 @@ type Receiver struct {
 	inFlight     inFlight                // used to track message disposition when rcv-settle-mode == second
 }
 
+// HandleMessage takes in a func to handle the incoming message.
+// Blocks until a message is received, ctx completes, or an error occurs.
+
+// Note: You must take an action on the message in the provided handler (Accept/Reject/Release/Modify)
+// or the pending message tracker will get out of sync, and reduce the flow.
+func (r *Receiver) HandleMessage(ctx context.Context, handle func(*Message) error) error {
+	debug(3, "Entering link %s Receive()", r.link.key.name)
+
+	trackCompletion := func(msg *Message) {
+		<-msg.doneSignal
+		r.link.deletePending(msg)
+		debug(3, "Receive() deleted pending %d", msg.deliveryID)
+		if atomic.LoadUint32(&r.link.paused) == 1 {
+			select {
+			case r.link.receiverReady <- struct{}{}:
+				debug(3, "Receive() unpause link on completion")
+			default:
+			}
+		}
+	}
+	callHandler := func(msg *Message) error {
+		debug(3, "Receive() blocking %d", msg.deliveryID)
+		msg.receiver = r
+		if msg.doneSignal == nil {
+			msg.doneSignal = make(chan struct{})
+		}
+		go trackCompletion(msg)
+		// tracks messages until exiting handler
+		if err := handle(msg); err != nil {
+			debug(3, "Receive() blocking %d - error: %s", msg.deliveryID, err.Error())
+			return err
```

Comment:
should we call `msg.done()` here?
I *think* it's not necessary since a handler error closes the connection and stops the listener entirely.